### DISCUSSION TITLE
how to get the current focused td

### DISCUSSION THREAD

**Discussion by SunnyCheung-cp** [Milkdown/milkdown#1116] [Q&A] [answered]
I want to add an options menu on td
I referenced the TableWidget,  I don't know how to get the current focused td

https://github.com/Milkdown/milkdown/assets/105030733/fec95ad3-0ce1-4f4e-84c8-2e062a5311f0

**Comment by Saul-Mirone** [comment]
You can get the rect by
```ts
import { selectedRect } from '@milkdown/prose/tables';

editor.action((ctx) => {
  const state = ctx.get(editorStateCtx);
  // See also:
  // https://github.com/ProseMirror/prosemirror-tables/blob/b569c2f9f63cb27eed6ada0bfd51ff434a03213b/src/commands.ts#L30
  const rect = selectedRect(state);
});
```

**Reply by SunnyCheung-cp** [reply, marked as answer]
@Saul-Mirone  Thank you for your help, It looks great

https://github.com/Milkdown/milkdown/assets/105030733/4a17dd52-0ff0-4de2-bcf3-f83a923b3836
